userdetails-plugin/grails-app/views/registration/createAccount.gsp

```diff
@@ -410,6 +424,66 @@
              document.getElementById("mfa").hidden = true
          });
 
+         $("#requestCode").click(function(e) {
+             var newEmail = $("#newEmail").val();
+             if(newEmail == null || newEmail === "") {
+                 document.getElementById("newEmailMessage").innerHTML = "Invalid email"
+                 document.getElementById("newEmailMessage").style.color = "red"
+                 document.getElementById("newEmailMessage").hidden = false
+             }
+             else {
+                 $.ajax({
+                 url: "${createLink(action:'update', controller: 'Registration')}?email=" + newEmail,
+                 type: "GET",
+                 success: function(result){
+                     let error = "A user is already registered with the email address"
+                     if(result.includes("Failed to update user profile")){
+                         if(result.includes(error)){
+                             document.getElementById("newEmailMessage").innerHTML = error
+                         }
+                         else {
+                            document.getElementById("newEmailMessage").innerHTML = 'Error please contact us at <a href="mailto:support@ala.org.au">support@ala.org.au</a>'
+                         }
+                         document.getElementById("newEmailMessage").style.color = "red"
+                         document.getElementById("newEmailMessage").hidden = false
+                         document.getElementById("emailCodeDiv").hidden = true
+                    }
+                     else{
+                        document.getElementById("newEmailMessage").innerHTML = "Please enter the code received in your new email"
+                        document.getElementById("newEmailMessage").style.color = "green"
+                        document.getElementById("newEmailMessage").hidden = false
+                        document.getElementById("emailCodeDiv").hidden = false
+                     }
+                }});
+             }
+         });
+
+         $("#verifyCode").click(function(e) {
+            var emailCode = $("#emailCode").val();
+             if(emailCode == null || emailCode === "") {
+                 document.getElementById("newEmailMessage").innerHTML = "Invalid code"
+                 document.getElementById("newEmailMessage").style.color = "red"
+                 document.getElementById("newEmailMessage").hidden = false
+                 document.getElementById("emailCodeDiv").hidden = false
+             }
+             else {
+                 $.ajax({
+                 url: "${createLink(action:'verifyAttributeChangeWithCode', controller: 'Registration')}?attribute=email&code=" + emailCode,
+                 type: "GET",
+                 success: function(result){
+                     if(result.success){
+                        window.location = "${createLink(uri:'/logout')}"
+                    }
```

Comment:
Instead of redirecting to logout (which after logging the user out redirects the ALA home page), it might be better to redirect to the login instead.

Reply:
Thanks Bruce. I will fix that.

Reply:
Fixed.